addon/components/o-s-s/tag.ts

```diff
@@ -0,0 +1,58 @@
+import Component from '@glimmer/component';
+
+export type SkinType = 'primary' | 'success' | 'warning' | 'danger' | 'secondary';
+
+type SkinDefType = {
+  [key in SkinType]: string;
+};
+const BASE_CLASS = 'upf-tag';
+
+export const SkinDefinition: SkinDefType = {
+  primary: 'regular',
+  success: 'success',
+  warning: 'alert',
+  danger: 'critical',
+  secondary: 'neutral'
+};
+
+interface OSSTagArgs {
+  label?: string;
+  skin?: string;
+  icon?: string;
+  hasEllipsis?: boolean;
+}
+
+export default class OSSTag extends Component<OSSTagArgs> {
+  constructor(owner: unknown, args: OSSTagArgs) {
+    super(owner, args);
+
+    if (!args.icon && !args.label) {
+      throw new Error('[component][OSS::Tag] You must pass either a @label or an @icon argument.');
+    }
+  }
+
+  get tooltip() {
+    return this.args.hasEllipsis ? this.args.label : null;
+  }
+
+  get skin(): string {
+    if (!this.args.skin) {
+      return SkinDefinition.primary;
+    }
```

Comment:
You can remove this as you have a default value return underneath if SkinDefinition[this.args.skin] does not match anything